geomstats/geometry/discrete_curves.py

```diff
@@ -326,6 +331,47 @@ def square_root_velocity_inverse(self, srv, starting_point):
 
         return curve
 
+    def d_square_root_velocity(self, tangent_vec, curve):
+        """Compute differential of the square root velocity transform.
+        """
+        if not isinstance(self.ambient_metric, EuclideanMetric):
+            raise AssertionError('The differential of the square root '
+                                 'velocity function is only implemented for '
+                                 'discrete curves embedded in a Euclidean '
+                                 'space.')
+        n_sampling_points = curve.shape[-2]
+        d_vec = n_sampling_points * (tangent_vec[1:, :] - tangent_vec[:-1, :])
+        velocity_vec = n_sampling_points * (curve[1:, :] - curve[:-1, :])
+        velocity_norm = self.ambient_metric.norm(velocity_vec)
+        unit_velocity_vec = gs.einsum(
+            'ij,i->ij', velocity_vec, 1 / velocity_norm)
+        d_vec_tangential = gs.einsum(
+            'ij,i->ij', unit_velocity_vec,
+            self.ambient_metric.inner_product(d_vec, unit_velocity_vec))
+        d_srv_vec = d_vec - 1 / 2 * d_vec_tangential
+        d_srv_vec = gs.einsum(
+            'ij,i->ij', d_srv_vec, 1 / velocity_norm**(1 / 2))
+
+        return d_srv_vec
+
+    def inner_product(self, tangent_vec_a, tangent_vec_b, curve):
```

Comment:
Why not calling `srv_inner_product` here?

Additionally, I wonder if there would be something interesting to be done with `PullbackMetric`:
https://github.com/geomstats/geomstats/blob/master/geomstats/geometry/pullback_metric.py

But maybe let's keep this discussion for another time.

Reply:
This is different from `srv_inner_product`:  `inner_product` is the actual inner product for the SRV metric, and can be computed for any two tangent vectors, i.e. infinitesimal vector fields along a same curve. The name of `srv_inner_product` is confusing (and in my opinion should be changed), as it simply computes the L2 inner product between the SRV representations of two curves. This is useful for computing distances between curves, but not inner products between tangent vectors. 

I haven't looked at `PullbackMetric` yet, but I will !

Reply:
Never mind, now I see what you meant and you are right: I can use `srv_inner_product` inside `inner_product`. Thanks !